cmd/aries-agentd/main_test.go

```diff
@@ -19,8 +20,25 @@ import (
 	"github.com/stretchr/testify/require"
 )
 
-const testURL = "localhost:8080"
-const testInboundURL = "localhost:8081"
+// nolint:gochecknoglobals
+var (
+	testURL        = fmt.Sprintf("localhost:%d", getRandomPort(false))
+	testInboundURL = fmt.Sprintf("localhost:%d", getRandomPort(false))
+)
+
+func getRandomPort(exit bool) int {
```

Comment:
I see we are recursing here. Can [this](https://github.com/hyperledger/aries-framework-go/blob/master/pkg/didcomm/transport/http/support_test.go#L107) be used here to generate the dynamic port number ?

Reply:
Do you want me to make getServerPort public and use it instead of `listener.Addr().(*net.TCPAddr).Port`?

Reply:
Now that I think about it again, I see this is different than the link from first post. Looks like we are creating a port and closing it, assuming it won't be taken by other processes before we start the server in main_test.go. There would be still some gap, but surely better than having static port.

Reply:
I assume this could have been structured as a loop with a number of attempts counter; rather than recursion.